presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -457,7 +457,7 @@ private void destroyIfNecessary()
         }
     }
 
-    private static ListenableFuture<?> isBlocked(Operator operator)
+    private static ListenableFuture<?> getBlockedFuture(Operator operator)
     {
         ListenableFuture<?> blocked = operator.isBlocked();
         if (blocked.isDone()) {
```

Comment:
I actually fear this can be a kind of memory leak. If a pipeline is something like aggregation and join probe side, Driver won't block because every time it has some page to move into aggregation, but the join probe is blocked and every time we'd add yet another future listener to this join's blocked future.